### PR TITLE
bazel as the build sytem

### PR DESCRIPTION
**Pull request by tburgin**

### CODE REVIEW

File: Restor/Controllers/ViewControllers/MainViewController.h [hunk 1]
```diff
@@ -12,15 +12,15 @@
...
 
 @class ConfigController;
 @class MOLXPCConnection;
 
 /// MainViewController mananges the main window containing the collection view,
 /// image selection and image/auto-image buttons.
-@interface MainViewController : NSViewController <NSCollectionViewDelegate,
-                                                  NSCollectionViewDataSource>
+@interface MainViewController
+    : NSViewController <NSCollectionViewDelegate, NSCollectionViewDataSource>
```
Comment (russellhancox) [inline Restor/Controllers/ViewControllers/MainViewController.h]: Ewww

File: Restor/Utilities/HelperToolUtils.m [hunk 2]
```diff
@@ -92,18 +92,16 @@ + (NSInteger)embeddedHelperVersion {
...
-                              kAuthorizationFlagExtendRights);
+  AuthorizationFlags flags = (kAuthorizationFlagDefaults | kAuthorizationFlagInteractionAllowed |
+                              kAuthorizationFlagPreAuthorize | kAuthorizationFlagExtendRights);
 
   NSString *promptText = @"Restor needs to install a helper tool.\n";
-  AuthorizationItem dialogConfiguration[1] = {
-      kAuthorizationEnvironmentPrompt, promptText.length, (char *)[promptText UTF8String], 0 };
-  AuthorizationEnvironment authorizationEnvironment = { 1, dialogConfiguration };
+  AuthorizationItem dialogConfiguration[1] = {kAuthorizationEnvironmentPrompt, promptText.length,
+                                              (char *)[promptText UTF8String], 0};
```
Comment (russellhancox) [inline Restor/Utilities/HelperToolUtils.m]: Also eww

File: Restor/Utilities/HelperToolUtils.m [hunk 3]
```diff
@@ -120,8 +118,8 @@ + (BOOL)installHelperToolWithError:(NSError **)error {
   // This installs the helper tool in /Library/PrivilegedHelperTools and the launchd.plist
   // in /Library/LaunchDaemons, then loads the job.
   CFErrorRef cfErrorRef;
-  BOOL result = SMJobBless(
-      kSMDomainSystemLaunchd, (__bridge CFStringRef)kHelperToolLabel, authRef, &cfErrorRef);
+  BOOL result = SMJobBless(kSMDomainSystemLaunchd, (__bridge CFStringRef)kHelperToolLabel, authRef,
+                           &cfErrorRef);
```
Comment (russellhancox) [inline Restor/Utilities/HelperToolUtils.m]: This too

File: WORKSPACE [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+apple_rules_dependencies()
+
+
+#Macops MOL* dependencies
+git_repository(
+    name = "MOLAuthenticatingURLSession",
+    remote =
+        "https://github.com/google/" +
+        "macops-molauthenticatingurlsession.git",
```
Comment (russellhancox) [inline WORKSPACE]: This is odd style?

File: restord/ImageSessionServer.m [hunk 5]
```diff
@@ -234,17 +234,10 @@ - (int)applyImage {
...
-                          @"--target",
-                          self.destination.path,
-                          @"--erase",
-                          @"--noprompt",
-                          @"--noverify",
-                          @"--puppetstrings" ];
+  self.asr.arguments = @[
+    @"restore", @"--buffersize", @"16m", @"--source", path, @"--target", self.destination.path,
+    @"--erase", @"--noprompt", @"--noverify", @"--puppetstrings"
+  ];
```
Comment (russellhancox) [inline restord/ImageSessionServer.m]: The older style was easier to read, imo.

File: restord/ImageSessionServer.m [hunk 6]
```diff
@@ -301,8 +294,7 @@ - (void)processOutput:(NSData *)output {
       NSLog(@"%@ %@", self, line);
     } else if ([line isEqualToString:@"Inverting target volume..."]) {
       [[self.client remoteObjectProxy] invertingStarted];
-    } else if (line.length >= dirtyLine.length &&
-               ![line isEqualToString:@"done"] &&
+    } else if (line.length >= dirtyLine.length && ![line isEqualToString:@"done"] &&
```
Comment (russellhancox) [inline restord/ImageSessionServer.m]: Likewise

File: signing_certificate.bzl [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    )
+
+expand_signing_certificate = rule(
+    attrs = {
+        "src": attr.label(allow_single_file = True),
+    },
+    fragments = ["objc"],
+    outputs = {"out": "%{name}.plist"},
+    implementation = _expand_signing_certificate,
+)
```
Comment (russellhancox) [inline signing_certificate.bzl:19]: Awesome

**Review by russellhancox — CHANGES_REQUESTED** [review]

**Comment by tburgin** [comment]
clang-format has some odd ideas about readability
I mainly wanted to import ordering. Going to revert some of the egregious chunks 😋

**Comment by tburgin** [comment]
PTAL

**Review by russellhancox — APPROVED** [review]

**Pull request merged**